### DISCUSSION TITLE
MAC OS terminal history contains weird temporary files

### DISCUSSION THREAD

**Discussion by hrz0x** [ohmyzsh/ohmyzsh#10369] [Help]
My ZSH history contains all these weird temporary folders like below. 
Does anyone know how to disable these?

`. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/m8IpeCGoNOkn'`
`. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/WMulDv81nQ6z'`

**Comment by rwmitchell** [comment]
Why do you think they are zsh history files?

From my quick check, they seem to be from a wide variety of applications and more likely is something baked deep into what the OS is doing.

**Reply by hrz0x** [reply]
> Why do you think they are zsh history files?
> 
> From my quick check, they seem to be from a wide variety of applications and more likely is something baked deep into what the OS is doing.

They appear in my zsh history files and also appear when I use the up arrow key to scroll through my terminal history in my terminal

**Comment by rwmitchell** [comment]
They are most likely in history files then because you (or someone) previously used your shell to explore those directories.

What exactly is in your history files?

**Reply by hrz0x** [reply]
The lines like the ones I posted on the main thread.

**Comment by rwmitchell** [comment]
that isn't the format of a zsh history file, try again.
Along with it, include lines before and after for context.

**Reply by hrz0x** [reply]
> that isn't the format of a zsh history file, try again. Along with it, include lines before and after for context.

Sorry. I wasn't being clear. When I open the `.zsh_history` file hundreds of lines like these are there between my normal commands.
```
: 1634952536:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/8gGGpFIb2VsV'
: 1634952584:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/loJ0qtajq3MX'
: 1634952665:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/IuUqBWL3BqEq'
: 1634952767:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/BFHAwkulBR5l'
: 1634952792:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/k3lQT0jmgApS'
: 1634952847:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/3jcAWG7NUArW'
: 1634952921:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/Yjw99Dt4tJ8n'
: 1634953038:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/BRA50wBhrUIl'
: 1634953208:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/FKOD2FJQHtkI'
: 1634953270:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/WTKsswxrJdEW'
: 1634953307:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/d3c3NWerg27R'
: 1634953345:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/l20VleS78Sv5'
: 1634953366:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/2n2nACnYYgI2'
: 1634953699:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/9QJEfsPQy628'
: 1634953875:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/AlcUFgUrHN1P'
: 1634953887:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/M0NRdbNMCUFW'
: 1634953927:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/37FYVXeJLNrR'
: 1634953944:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/EOK6b3tUR9ZN'
: 1634953980:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/eAvkp6jrymZn'
: 1634954037:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/x2Wje4H9XTC8'
: 1634954044:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/2WrvvBafpFTA'
: 1634954088:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/AVy099ILuHbA'
: 1634954159:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/60mwkOR3aiiF'
: 1634954185:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/15FWh9leY0bn'
: 1634954204:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/YCNFHsBEEpBG'
: 1634954264:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/SWor2ZqGAq0y'
: 1634954340:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/1pipdZ7BCY0e'
: 1634954432:0;. '/var/folders/tq/spbb_x2s57x5nr57d7d8v7980000gn/T/jueW9LULeP9d'
```

**Comment by rwmitchell** [comment]
The leading dot suggests it (something) was trying to source those files/directories.  Have you taken a look to see what any of them are?  Again, from what I know, those are normally just from OS X housekeeping / library calls.  I have no idea why they would be in your history - can't find any evidence of them being in mine.  Is there anything special/common in the command that precedes the start of these line?  What plugins are you using?

```echo $plugins | tr ' ' '\n'```

**Reply by hrz0x** [reply]
git
osx
web-search

No they appear at almost every random command. Most of my commands are just things like python3 blahblah.py and rclone etc.  I think it is something related to the clipboard since those occur a lot especially when I am using clipboards.

**Comment by mcornella** [comment]
Without more information it's impossible to know what it is. It looks like something is running `. =(<command>)` or something like that. If you post your zshrc file we could take a look.
